spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/GqlDatastoreQuery.java

```diff
@@ -132,6 +146,13 @@ public Object execute(Object[] parameters) {
 		return result;
 	}
 
+	private String getGqlResolvedEntityClassName() {
+		if (this.gqlResolvedEntityClassName == null) {
```

Comment:
maybe move caching into `resolveEntityClassNames(this.originalGql)` method and remove this one?